src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Applications/ApplicationsEndpoints.cs

```diff
@@ -30,12 +30,29 @@ public void Register(IEndpointRouteBuilder endpointRouteBuilder)
         .RequireAuthorization()
         .WithParameterValidation();
 
-    endpointRouteBuilder.MapPost("/api/applications", async (ApplicationSubmissionRequest request, IMessageBus messageBus) =>
+    endpointRouteBuilder.MapPost("/api/applications", async Task<Results<Ok<SubmitApplicationResponse>, BadRequest<string>>> (ApplicationSubmissionRequest request, HttpContext ctx, IMessageBus messageBus) =>
         {
-          var cmd = new SubmitApplicationCommand(request.Id);
-          await messageBus.InvokeAsync<string>(cmd);
+          var userId = ctx.User.GetUserContext()?.UserId;
+          bool IdIsNotGuid = !Guid.TryParse(request.Id, out _); if (IdIsNotGuid) { return TypedResults.BadRequest("ApplicationId is not valid"); }
+          var query = new ApplicationsQuery
+          {
+            ById = request.Id,
+            ByApplicantId = userId,
+            ByStatus = new List<Managers.Registry.Contract.Applications.ApplicationStatus>() { Managers.Registry.Contract.Applications.ApplicationStatus.Draft }
+          };
+          var results = await messageBus.InvokeAsync<ApplicationsQueryResults>(query);
+          if (!results.Items.Any())
+          {
+            return TypedResults.BadRequest("Application not found");
+          }
 
-          return TypedResults.Ok();
+          var cmd = new SubmitApplicationCommand(request.Id);
+          var result = await messageBus.InvokeAsync<ApplicationSubmissionResult>(cmd);
```

Comment:
We need to go through the code and ensure we pass in the cancallation tokens all the way down, it will help the system to handle aborted connections and reduce load

Reply:
I implemented the CancellationToken for save and submit application functionalities. review that and see if it follows the right pattern here.